### ISSUE TITLE
Support for UDP ports?

### ISSUE THREAD

**Issue by KallynGowdy** [rancher-sandbox/rancher-desktop-agent#3]
It appears that this agent currently only support setting up forwards for TCP listeners. Would it be possible to detect and support UDP ports as well?

The [iptables lookup entries](https://github.com/lima-vm/lima/blob/d87e89f7ea651cc2e17fc3a913eb62b77c07e8d6/pkg/guestagent/iptables/iptables.go#L14) already include whether the port is TCP/UDP, so hopefully it is as simple as checking the flag and opening the listener in a different mode.

Thoughts?

**Comment by jandubois** [comment]
> hopefully it is as simple as checking the flag and opening the listener in a different mode.

I don't think the network bridge between Windows and WSL support UDP port forwarding, but am not sure.

We'll eventually implement our own port forwarding code to also allow listening on external interfaces, and not just on localhost.

@Nino-K should have more details.

**Comment by KallynGowdy** [comment]
Yeah, it looks like this is a limitation of WSL. They've been using [this issue](https://github.com/microsoft/WSL/issues/4825) as the basis point for several UDP-related problems, but it doesn't look like there's much activity on that issue specifically. There's a lot more discussion about [supporting bridge networking](https://github.com/microsoft/WSL/issues/4150) with WSL (and even [some support for it in the WSL Preview](https://github.com/microsoft/WSL/issues/4150#issuecomment-1135474390)), but it is unclear what that would do for UDP networking or even if it is an acceptable solution.

**Comment by jandubois** [comment]
We have a plan to tunnel all traffic through a vsock so it looks like it originates from and terminates on the host, and not inside WSL2. That should also help with proxy settings and other problems, and we will be decoupled from whatever WSL2 is doing. But it will take a while to get implemented.
